src/wrappers/themis/react-native-themis/android/src/main/java/com/reactnativethemis/ThemisModule.java

```diff
@@ -33,608 +32,544 @@
-    try {
-      enc = dataDeserialize(encrypted);
-    } catch (ByteOverflowException e) {
-      errorCallback.invoke(e.getMessage());
-      return;
-    }
-
-    byte[] ctx = context.getBytes(StandardCharsets.UTF_8);
-    try {
-      byte[] decrypted = cell.decrypt(enc, ctx);
-      WritableArray response = dataSerialize(decrypted);
-      successCallback.invoke(response);
-    } catch (SecureCellException e) {
-      errorCallback.invoke(e.getMessage());
-    }
-
-  }
-
-  @ReactMethod
-  public void secureCellSealWithPassphraseEncrypt(String passphrase,
-                                              String plaintext,
-                                              String context,
-                                              Callback callback)
-  {
-    SecureCell.Seal cell = SecureCell.SealWithPassphrase(passphrase);
-    byte[] txt = plaintext.getBytes(StandardCharsets.UTF_8);
-    byte[] ctx = context.getBytes(StandardCharsets.UTF_8);
-    byte[] encrypted = cell.encrypt(txt, ctx);
-    WritableArray response = dataSerialize(encrypted);
-    callback.invoke(response);
-  }
-
-  @ReactMethod
-  public void secureCellSealWithPassphraseDecrypt(String passphrase,
+    @NonNull
+    @Override
+    public String getName() {
+        return "Themis";
+    }
+
+    /* Replace JAVA enum version to equal with iOS version. */
+
+    @Override
+    public Map<String, Object> getConstants() {
+        final Map<String, Object> constants = new HashMap<>();
+        constants.put("COMPARATOR_NOT_READY", comparator_not_ready);
+        constants.put("COMPARATOR_NOT_MATCH", comparator_not_match);
+        constants.put("COMPARATOR_MATCH", comparator_match);
+        constants.put("COMPARATOR_ERROR", comparator_error);
+        constants.put("KEYTYPE_RSA", AsymmetricKey.KEYTYPE_RSA);
+        constants.put("KEYTYPE_EC", AsymmetricKey.KEYTYPE_EC);
+        return constants;
+    }
+
+    private static int compareResultSerialize(SecureCompare.CompareResult result) {
+        switch (result) {
+        case NOT_READY:
+            return comparator_not_ready;
+        case NO_MATCH:
+            return comparator_not_match;
+        case MATCH:
+            return comparator_match;
+        default:
+            return comparator_error;
+        }
+    }
+
+    private static WritableArray dataSerialize(byte[] data) {
+        if (data == null) {
+          return null;
+        }
+        WritableArray result = new WritableNativeArray();
+        for (byte datum : data)
+            result.pushInt(datum);
+
+        return result;
+    }
+
+    private static byte[] dataDeserialize(ReadableArray serializedData) throws ByteOverflowException {
+        if (serializedData == null || serializedData.size() == 0) {
+            return null;
+        }
+        byte[] data = new byte[serializedData.size()];
+        for (int i = 0; i < serializedData.size(); i++) {
+            if (serializedData.getInt(i) >= -128 && serializedData.getInt(i) <= 255) {
```

Comment:
why we check range -128..255 instead of -128..127? do we have any tests of these checks? to be sure that it doesn't accept overflowed integers.

Reply:
It is a historical issue. The correct value is 0...255. I fixed it in the update.